packages/mdx/mdx-ast-to-mdx-hast.js

```diff
@@ -18,6 +20,38 @@ module.exports = function mdxAstToMdxHast() {
           ]
         })
       },
+      code(h, node) {
+        var value = node.value ? detab(node.value + '\n') : ''
+        var lang = node.lang && node.lang.match(/^[^ \t]+(?=[ \t]|$)/)
+        var props = {}
+
+        if (lang) {
+          props.className = ['language-' + lang]
+        }
+
+        props.metaString =
+          node.lang && node.lang.replace(/^[^ \t]+(?=[ \t]|$)/, '').trim()
+
+        const meta =
+          props.metaString &&
+          props.metaString.split(' ').reduce((acc, cur) => {
+            if (cur.split('=').length > 1) {
+              const t = cur.split('=')
+              acc[t[0]] = t[1]
+              return acc
+            } else {
+              acc[cur] = true
+              return acc
+            }
+          }, {})
+
+        meta &&
+          Object.entries(meta).forEach(([key, value]) => (props[key] = value))
```

Comment:
Not sure we should drop node 8 support until we go v1, maybe we can make this a util or use `Object.keys`?